.github/workflows/release.yaml

```diff
@@ -1,64 +1,80 @@
+    paths:
+      - '.github/project.yaml'
 jobs:
-  build-release:
-    name: Create Release
-    runs-on: ubuntu-latest
+  release:
+    runs-on: ubuntu-18.04
+    if: ${{github.event.pull_request.merged == true}}
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v2
-      - name: Set up Node.js
+      - name: Retrieve Project Metadata
+        uses: radcortez/project-metadata-action@master
+        id: metadata
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          metadata-file-path: '.github/project.yaml'
+      - name: Set up Node.js v12
         uses: actions/setup-node@v1
         with:
           node-version: 12
       - name: Set up JDK 1.8
         uses: actions/setup-java@v1
         with:
           java-version: 1.8
+      - name: Set up settings.xml
+        run: |
+          echo "<settings><servers><server><id>${{ secrets.OSSRH_ID }}</id><username>${{ secrets.OSSRH_USERNAME }}</username><password>${{ secrets.OSSRH_TOKEN }}</password></server></servers><profiles><profile><id>${{ secrets.OSSRH_ID }}</id><activation><activeByDefault>true</activeByDefault></activation><properties><gpg.executable>gpg</gpg.executable><gpg.passphrase>${{ secrets.GPG_PASSPHRASE}}</gpg.passphrase></properties></profile></profiles></settings>" > /home/runner/.m2/settings.xml
       - name: Set up Gren
-        if: github.event_name == 'repository_dispatch'
         run: npm install github-release-notes -g
       - name: Cache Dependencies
-        if: github.event_name != 'repository_dispatch'
         uses: actions/cache@v1
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-maven-
-      - name: Build Project
-        if: github.event_name != 'repository_dispatch'
-        run: mvn clean install
-      - name: Setup settings.xml
-        if: github.event_name == 'repository_dispatch'
+      - name: Code Checkout
         run: |
-          echo "<settings><servers><server><id>${{ secrets.OSSRH_ID }}</id><username>${{ secrets.OSSRH_USERNAME }}</username><password>${{ secrets.OSSRH_TOKEN }}</password></server></servers><profiles><profile><id>${{ secrets.OSSRH_ID }}</id><activation><activeByDefault>true</activeByDefault></activation><properties><gpg.executable>gpg</gpg.executable><gpg.passphrase>${{ secrets.GPG_PASSPHRASE}}</gpg.passphrase></properties></profile></profiles></settings>" > /home/runner/.m2/settings.xml
+          git init
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git remote add origin "https://$GITHUB_ACTOR:${{ secrets.ACCESS_TOKEN }}@github.com/$GITHUB_REPOSITORY.git"
+          git fetch
+          git checkout master
+          git branch --set-upstream-to=origin/master
+          git pull
+          echo "#### Listing files after clone ####"
+          ls -a
```

Comment:
Why the added complexity here vs. just a standard `git clone`?

Reply:
Or I guess, why do this instead of:
```
      - name: Checkout Code
        uses: actions/checkout@v2
```

?

Reply:
Since, the virtual machine on which the workflow runs doesn't have ssh key configured, we get permission error when try to do a push or pull. Since we are doing push to the master, which is protected by rules, the default token used by the checkout action doesn't have sufficient permission. This way I am passing my custom access token @line 44 and get myself authenticated. this way even if i do a push or pull in other steps as well i do not get a permission error

Reply:
OK thanks for the explanation.  I think we should use the `apicurio-ci` user account for this, do you agree?  I am waiting for GitHub to unflag that account so that we can use it for this purpose.

Reply:
Will the user have sufficient privileges to do a push to the rules protected master branch ?

Reply:
We'll make sure that user has the appropriate privs.